webapp/channels/src/utils/url.tsx

```diff
@@ -168,6 +168,15 @@ export function validateChannelUrl(url: string, intl?: IntlShape): Array<React.R
     return errors;
 }
 
+// Returns true when the URL should not trigger any external requests.
+// Currently internal links that are not to the image proxy.
+export function doesNotTriggerExternalRequests(url: string, siteURL?: string): boolean {
+    if (siteURL && siteURL !== '') {
+        return url.startsWith(siteURL) && !url.includes('v4/image');
```

Comment:
`v4//image`, `v4\image`, `v4/./image`, `v4 /image`... ? And the image proxy isn't even the only unsafe endpoint, as you've noted yourself: https://github.com/mattermost/mattermost/pull/25889/files#r1467830234

Reply:
True. Fixed to allow only permalinks.